src/Microsoft.Health.Fhir.Core/Features/Operations/Export/ExportJobTask.cs

```diff
@@ -365,7 +365,10 @@ private async Task ProcessSearchResultsAsync(IEnumerable<SearchResultEntry> sear
                     else
                     {
                         // File does not exist. Create it.
-                        string fileName = resourceType + ".ndjson";
+                        string dateTime = _exportJobRecord.QueuedTime.UtcDateTime.ToString("s")
+                            .Replace("-", string.Empty, StringComparison.OrdinalIgnoreCase)
+                            .Replace(":", string.Empty, StringComparison.OrdinalIgnoreCase);
+                        string fileName = $"{dateTime}-{_exportJobRecord.Id}/{resourceType}.ndjson";
```

Comment:
Anyone who has already setup an automated workflow for export will now be broken by this change (even though they are not using the `_container` parameter for their export requests). Should we try to maintain backward compatibility and only add the folder if the customer supplies `_container` parameter? If we are fine with making this breaking change, we might want to think about how we will notify existing users.

Reply:
@brendankowitz? I think that the change is needed to keep the containers organized incase a user decides to reuse one of the GUID containers we make, and to keep the internal structure consistent between export jobs. I don't know who is the PM for OSS at this point.

Reply:
I had an offline discussion with RB on this. We can go with this: If the customer does not provide the container name, we export as per the prior behavior. If the customer provides the container name, we create folder inside the container for the export. We should document this behavior in user documentation.